axum/src/extract/ws.rs

```diff
@@ -267,14 +267,15 @@ where
             return Err(InvalidWebSocketVersionHeader.into());
         }
 
-        let sec_websocket_key =
-            if let Some(key) = req.headers_mut().remove(header::SEC_WEBSOCKET_KEY) {
-                key
-            } else {
-                return Err(WebSocketKeyHeaderMissing.into());
-            };
+        let sec_websocket_key = req
+            .headers_mut()
+            .remove(header::SEC_WEBSOCKET_KEY)
+            .ok_or(WebSocketKeyHeaderMissing)?;
 
-        let on_upgrade = req.extensions_mut().remove::<OnUpgrade>().unwrap();
+        let on_upgrade = req
+            .extensions_mut()
+            .remove::<OnUpgrade>()
+            .ok_or(MissingUpgradeState)?;
```

Comment:
@seanmonstar How do you recommend axum deals with `OnUpgrade` missing? We got this issue https://github.com/tokio-rs/axum/issues/1132 recently from a user who ran into this.

This code is inspired by warp which just logs if the extension is missing (https://github.com/seanmonstar/warp/blob/master/src/filters/ws.rs#L154).

Is is possible to get more details from hyper about why the upgrade failed?